spec/controllers/queue_items_controller_spec.rb

```diff
@@ -4,130 +4,114 @@
+      queue_item1 = Fabricate(:queue_item, user: current_user)
+      queue_item2 = Fabricate(:queue_item, user: current_user)
       get :index
       expect(assigns(:queue_items)).to match_array([queue_item1, queue_item2])
     end
 
-    it "redirects the user to sign in page for unauthenticated user" do
-      get :index
-      expect(response).to redirect_to sign_in_path
+    context "with unauthenticated user" do
+      it_behaves_like "require_sign_in" do
+        let(:action) { get :index }
+      end
     end
   end
 
   describe "POST create" do 
     context "with authenticated user" do
 
-      let(:thupten) { Fabricate(:user) }
-
-      before { session[:user_id] = thupten.id }
+      before { set_current_user }
+      let(:monk) { Fabricate(:video) }
 
       it "redirects to my queue page" do 
-        video = Fabricate(:video)
-        post :create, video_id: video.id
+        post :create, video_id: monk.id
         expect(response).to redirect_to my_queue_path
       end
 
       it "creates a queue_item that is associated with a video" do
-        video = Fabricate(:video)
-        post :create, video_id: video.id
+        post :create, video_id: monk.id
         expect(QueueItem.count).to eq(1)
       end
 
       it "creates a queue_item that is associated with a video" do
-        video = Fabricate(:video)
-        post :create, video_id: video.id
-        expect(QueueItem.first.video).to eq(video)
+        post :create, video_id: monk.id
+        expect(QueueItem.first.video).to eq(monk)
       end
 
       it "creates the queue_item that is associated with a user" do
-        video = Fabricate(:video)
-        post :create, video_id: video.id
-        expect(QueueItem.first.user).to eq(thupten)
+        post :create, video_id: monk.id
+        expect(QueueItem.first.user).to eq(current_user)
       end
 
       it "puts the new queue_item at the bottom of the queue_items list" do
-        monk = Fabricate(:video)
-        Fabricate(:queue_item, video: monk, user: thupten)
+        Fabricate(:queue_item, video: monk, user: current_user)
         southpark = Fabricate(:video)
         post :create, video_id: southpark.id
-        southpark_queue_item = QueueItem.where(video_id: southpark.id, user_id: thupten.id).first
+        southpark_queue_item = QueueItem.where(video_id: southpark.id, user_id: current_user.id).first
         expect(southpark_queue_item.position).to eq(2)
       end
 
       it "does not add the same video twice" do 
-        monk = Fabricate(:video)
-        Fabricate(:queue_item, video: monk, user: thupten)
+        Fabricate(:queue_item, video: monk, user: current_user)
         southpark = Fabricate(:video)
         post :create, video_id: monk.id
-        southpark_queue_item = QueueItem.where(video_id: southpark.id, user_id: thupten.id).first
-        expect(thupten.queue_items.count).to eq(1)
+        southpark_queue_item = QueueItem.where(video_id: southpark.id, user_id: current_user.id).first
```

Comment:
`where(args).first` should be just `.find_by` - that is the new standard in Rails 4